packages/playwright/src/transform/compilationCache.ts

```diff
@@ -218,9 +218,15 @@ export function collectAffectedTestFiles(dependency: string, testFileCollector:
     if (deps.has(dependency))
       testFileCollector.add(testFile);
   }
-  for (const [testFile, deps] of externalDependencies) {
-    if (deps.has(dependency))
-      testFileCollector.add(testFile);
+  for (const [importingFile, deps] of externalDependencies) {
```

Comment:
Let's also update `dependenciesForTestFile` function.

Reply:
I've updated this in https://github.com/microsoft/playwright/pull/31794/commits/72c446d8ad202dc60517c99be6755231b2352885 and verified manually that it works correctly, but i'm unsure how to add a test for it. The problem is that the function is only ever used to update the file watcher, and it's hard to observe that. Where would you add a test for this?

Reply:
I think we can live without a test for this for now.